http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpWsmDao.scala

```diff
@@ -153,6 +156,78 @@ class HttpWsmDao[F[_]](httpClient: Client[F], config: HttpWsmDaoConfig)(implicit
       )(onError)
     } yield res
 
+  override def getLandingZoneResources(billingProfileId: String, userToken: Authorization)(implicit
+    ev: Ask[F, AppContext]
+  ): F[LandingZoneResources] =
+    for {
+      // Step 1: call LZ for LZ id
+      landingZoneOpt <- getLandingZone(billingProfileId, userToken)
+      landingZone <- F.fromOption(
+        landingZoneOpt,
+        AppCreationException(s"Landing zone not found for billing profile ${billingProfileId}")
+      )
+      landingZoneId = landingZone.landingZoneId
+
+      // Step 2: call LZ for LZ resources
+      lzResourcesByPurpose <- listLandingZoneResourcesByType(landingZoneId, userToken)
+
+      aksClusterName <- getLandingZoneResourceName(lzResourcesByPurpose,
+        "Microsoft.ContainerService/managedClusters",
+        SHARED_RESOURCE,
+        false
+      )
+      batchAccountName <- getLandingZoneResourceName(lzResourcesByPurpose,
+        "Microsoft.Batch/batchAccounts",
+        SHARED_RESOURCE,
+        false
+      )
+      relayNamespace <- getLandingZoneResourceName(lzResourcesByPurpose,
+        "Microsoft.Relay/namespaces",
+        SHARED_RESOURCE,
+        false
+      )
+      storageAccountName <- getLandingZoneResourceName(lzResourcesByPurpose,
+        "Microsoft.Storage/storageAccounts",
+        SHARED_RESOURCE,
+        false
+      )
+      vnetName <- getLandingZoneResourceName(lzResourcesByPurpose, "DeployedSubnet", AKS_NODE_POOL_SUBNET, true)
+      batchNodesSubnetName <- getLandingZoneResourceName(lzResourcesByPurpose,
+        "DeployedSubnet",
+        WORKSPACE_BATCH_SUBNET,
+        true
+      )
+      aksSubnetName <- getLandingZoneResourceName(lzResourcesByPurpose, "DeployedSubnet", AKS_NODE_POOL_SUBNET, true)
+    } yield LandingZoneResources(
+      AKSClusterName(aksClusterName),
+      BatchAccountName(batchAccountName),
+      RelayNamespace(relayNamespace),
+      StorageAccountName(storageAccountName),
+      NetworkName(vnetName),
+      SubnetworkName(batchNodesSubnetName),
+      SubnetworkName(aksSubnetName)
+    )
+
+  private def getLandingZoneResourceName(landingZoneResourcesByPurpose: List[LandingZoneResourcesByPurpose],
+                                         resourceType: String,
+                                         purpose: LandingZoneResourcePurpose,
+                                         useParent: Boolean
+                                        ): F[String] =
+    landingZoneResourcesByPurpose
+      .filter(_.purpose == purpose)
+      .flatMap(_.deployedResources)
+      .filter(_.resourceType.equalsIgnoreCase(resourceType))
+      .headOption
+      .flatMap { r =>
+        if (useParent) r.resourceParentId.flatMap(_.split('/').lastOption)
+        else r.resourceName.orElse(r.resourceId.flatMap(_.split('/').lastOption))
+      }
+      .fold(
+        F.raiseError[String](
+          AppCreationException(s"${resourceType} resource with purpose ${purpose} not found in landing zone")
```

Comment:
the error probably should be updated?

What's a LZ resource name btw? can u give an example if u have it handy

Reply:
Yes, the names are autogenerated by the landing zone service based on the azure resource requirements (different resources get names of different lengths). Here's an example from logs of some names:
```
{
  "clusterName":"lz87b5acf237a7394344358a9",
  "batchAccountName":"lze5ac32cc7822076fc571a8",
  "relayNamespace":"lz1aad27142d64b2999b8ceae00ada4b3c9d2bf58a098ca336",
  "storageAccountName":"lz9b7e56e0d2d94c6031edc2",
   "vnetName":"lzf43fb0d47b61bd524490895ca52933b304f3ceec72736d76220753ef17bc8d",
  "batchNodesSubnetName":"lzf43fb0d47b61bd524490895ca52933b304f3ceec72736d76220753ef17bc8d",
  "aksSubnetName":"lzf43fb0d47b61bd524490895ca52933b304f3ceec72736d76220753ef17bc8d"
}
```